src/main/applied_science/js_interop.cljs

```diff
@@ -218,3 +218,17 @@
     (doseq [[k v] (partition 2 keyvals)]
       (j/assoc! obj k v))
     obj))
+
+(defn extend
+  "Extends `obj` with the properties of one or more objects, overwriting
+   existing properties, moving left to right through `objs`. Returns `obj`.
+   An empty starting object is provided if `obj` is nil.
+  ```
+  (j/extend o other)
+  (j/extend o other #js{:x 1})
+  ```"
+  [obj & objs]
+  (let [obj (if (some? obj) obj #js{})
+        args (conj objs obj)]
```

Comment:
not necessary to make an intermediate list here - can pass `obj` and `objs` directly to `core/apply` separately (eg. `(apply x obj objs)`)